File: 2-add-two-numbers/2-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val 
#         self.next = next 
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        headNode = None
        lastNode = headNode
        carry = 0
        while True:
            if l1 == None and l2 ==None:
                if carry == 1:
                    tmpNode = ListNode()
                    tmpNode.val = 1
                    tmpNode.next = None
                    lastNode.next = tmpNode
                return headNode
            else:
                if l1 == None:
                    v1 = 0
                    v1Next = None
                else:
                    v1 = l1.val
                    v1Next = l1.next
                    
                if l2 == None:
                    v2 = 0
                    v2Next = None
                else:
                    v2 = l2.val
                    v2Next = l2.next
                    
                tmpNode = ListNode()
                val = v1 + v2 + carry
                if val >=10:
                    tmpNode.val = val - 10
                    carry = 1
                else:
                    tmpNode.val = val
                    carry = 0
                    
                if headNode == None:
                    headNode = tmpNode
                    lastNode = headNode
                else:
                    lastNode.next = tmpNode
                    lastNode = tmpNode
                l1 = v1Next
                l2 = v2Next
                

File: 2-add-two-numbers/2-add-two-numbers.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
    carry := 0
    v1 := 0
    v2 := 0
    v1Next  := &ListNode{}
    v2Next  := &ListNode{}
    var headNode *ListNode
    var endNode *ListNode
    for true{
        //fmt.Println(l1)
        //fmt.Println(l2)
        //fmt.Println(headNode)
        if (l1 == nil && l2 == nil ) {
            if carry == 1{
                tmpNode := &ListNode{}
                tmpNode.Val = 1
                endNode.Next = tmpNode
            }
            //fmt.Println("end!!!")
            return headNode
        }
        //fmt.Println("loop again...")
        if ( l1 == nil ) {
            v1 = 0
            v1Next = nil
        }else{
            v1 = l1.Val
            v1Next = l1.Next
        }
        
        if ( l2 == nil ) {
            v2 = 0
            v2Next = nil
            
        }else{
            v2 = l2.Val
            v2Next = l2.Next
        }    
        
        tmpNode := &ListNode{}
        val := v1 + v2 + carry
        if val >= 10{
            tmpNode.Val = val - 10
            carry = 1
            
        }else{
            tmpNode.Val = val
            carry = 0
        }
        
        
        if ( headNode == nil ) {
            headNode = tmpNode
            endNode = headNode
        }else{
            endNode.Next = tmpNode
            endNode = tmpNode
        }
        
        l1 = v1Next
        l2 = v2Next
        //fmt.Println(headNode)
    }
    return headNode
}